url.go

```diff
@@ -143,9 +174,21 @@ func parseV1URL(url *gourl.URL, originalURL string) (*URL, error) {
 		}
 		r.User, parts = parts[0][1:], parts[1:]
 	}
+
+	// <channel>
+	if len(parts) > 1 {
+		if IsValidChannel(parts[0]) {
+			if r.Schema == "local" {
+				return nil, fmt.Errorf("local charm or bundle URL with channel: %q", originalURL)
```

Comment:
This doesn't read much like an error.

local charm or bundle URL %q is not allowed to have a channel

?